File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        number_map = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in number_map:
                return [number_map[complement], i]
            number_map[num] = i
        return []
        

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function twoSum(nums, target) {
    let numberMap = {};
    for (let i = 0; i < nums.length; i++) {
        let compliment = target - nums[i];
        if (numberMap.hasOwnProperty(compliment)) {
            return [numberMap[compliment], i];
        }
        numberMap[nums[i]] = i;
    }
}